src/components/CourseCard/CourseCard.tsx

```diff
@@ -0,0 +1,178 @@
+import classNames from 'classnames';
+import Hls from 'hls.js';
+import {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Link } from 'react-router-dom';
+
+import { useAppDispatch } from '../../app/hooks';
+import { selectedCourseInit } from '../../features/selectedCourseSlice';
+import { Courses } from '../../types/Courses';
+import { RatingStars } from '../RatingStars';
+import './CourseCard.scss';
+
+type Props = {
+  course: Courses,
+};
+
+export const CourseCard: React.FC<Props> = ({ course }) => {
+  const {
+    id,
+    tags,
+    title,
+    previewImageLink,
+    lessonsCount,
+    meta,
+    rating,
+  } = course;
+
+  const [isHovered, setIsHovered] = useState(false);
+
+  const videoRef = useRef<HTMLVideoElement | null>(null);
+
+  const handleMouseEnter = () => {
+    setIsHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovered(false);
+
+    if (videoRef.current && !videoRef.current.paused) {
+      videoRef.current.pause();
+    }
+  };
+
+  const dispatch = useAppDispatch();
+
+  const handleClick = () => {
+    dispatch(selectedCourseInit(id));
+  };
+
+  const handleOnKeyDown = (event: React.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      setIsHovered(false);
+    }
+
+    if (event.key === 'Enter') {
+      handleClick();
+    }
+  };
+
+  useEffect(() => {
+    let hls: Hls;
+
+    if (isHovered && videoRef.current) {
+      if (!videoRef.current.paused) {
+        videoRef.current.pause();
+      }
+
+      hls = new Hls({ enableWorker: false });
+
+      if (meta.courseVideoPreview?.link) {
+        hls.loadSource(meta.courseVideoPreview.link);
+        hls.attachMedia(videoRef.current);
+
+        hls.on(Hls.Events.MANIFEST_PARSED, () => {
+          videoRef.current?.play();
+        });
+      } else {
+        setIsHovered(false);
+      }
+    }
+
+    return () => {
+      if (hls) {
+        hls.destroy();
+      }
+    };
+  }, [isHovered, meta.courseVideoPreview?.link]);
+
+  return (
+    <Link
+      to={`../course/${id}`}
+      className="course-card"
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+      onKeyDown={handleOnKeyDown}
+      onClick={handleClick}
+    >
+      <h5
+        className={classNames(
+          'course-card__category',
+          { 'course-card__category--yellow': tags[0] === 'productivity' },
+          { 'course-card__category--purple': tags[0] === 'learning ability' },
+          { 'course-card__category--red': tags[0] === 'communication' },
+          { 'course-card__category--green': tags[0] === 'psychology' },
+        )}
```

Comment:
It seems to me such a long list of classnames is better to be passed as a separate var to increase a readability of code.